tests/main/generic-unregister/task.yaml

```diff
@@ -0,0 +1,44 @@
+summary: |
+    Ensure that unregistration API works.
+
+# ubuntu-14.04: curl does not have --unix-socket option
+systems: [-ubuntu-core-*, -ubuntu-14.04-*]
+
+prepare: |
+    systemctl stop snapd.service snapd.socket
+    cp /var/lib/snapd/state.json state.json.bak
+    mkdir key
+    cp /var/lib/snapd/device/private-keys-v1/* key
+    systemctl start snapd.service snapd.socket
+
+restore: |
+    systemctl stop snapd.service snapd.socket
+    cp key/* /var/lib/snapd/device/private-keys-v1/
+    cp state.json.bak /var/lib/snapd/state.json
+    systemctl start snapd.service snapd.socket
+
+execute: |
+    #shellcheck source=tests/lib/core-config.sh
+    . "$TESTSLIB"/core-config.sh
+
+    wait_for_device_initialized_change
+
+    snap model --assertion | MATCH "series: 16"
+
+    if snap model --verbose | NOMATCH "brand-id:\s* generic" ; then
+       echo "Not a generic model. Skipping."
+       exit 0
+    fi
+
+    curl --data '{"action":"forget","no-registration-until-reboot":true}' --unix-socket /run/snapd.socket http://localhost/v2/model/serial
+
+    test -f /run/snapd/noregister
+
+    snap model --serial 2>&1|MATCH "error: device not registered yet"
+
+    systemctl restart snapd.service
+
+    snap model --serial 2>&1|MATCH "error: device not registered yet"
+    snap find pc
```

Comment:
what's this line for? is this meant to trigger snapd to interact with the store and thus possibly to get registration if it wasn't told to block the registration until the next reboot?

Reply:
yes, something like that. tbh it's hard to prove that something is not being done but given we also know the code it seems alright